docs/ARCHITECTURE.md

```diff
@@ -22,21 +22,46 @@ effective tool for finding and removing individual records within objects stored
-   multiple matches to be removed in a single job.
-2. **Fail fast:** A deletion job takes place in two distinct phases: Find and
-   Forget. The Find phase queries the objects in your S3 data lakes to find any
-   objects which contain records where a specified column contains at least one
-   of record identifiers (known as **Match IDs**) in the deletion queue. If any
-   queries fail, the job will abandon as soon as possible and the Forget phase
-   will not take place. The Forget Phase takes the list of objects returned from
-   the Find phase, and deletes the relevant rows in only those objects.
-3. **Serverless:** Where possible, the solution only uses Serverless components.
-   All the components for Web UI, API and Deletion Jobs are Serverless (for more
-   information consult the [Cost Overview guide]).
+1. **Security by design:** Every component implements least privilege access,
+   encryption is performed at all layers at rest and in transit, authentication
+   is provided out of the box, expiration of logs is configurable, and record
+   identifiers (known as **Match IDs**) are automatically obfuscated or
+   irreversibly deleted as soon as possible when persisting state.
+2. **Built to scale:** The system has been designed and battle tested for
+   performance and high availability with Data Lakes of TB/PB size.
+3. **Cost optimised:**
+   - **Perform work in batches:** Since the time complexity of removing a single
+     vs multiple records in a single object is practically equal and it is
+     common for data owners to have the requirement of removing data within a
+     given _timeframe_, the solution is designed to allow the solution operator
+     to "queue" multiple matches to be removed in a single job.
+   - **Fail fast:** A deletion job takes place in two distinct phases: Find and
+     Forget. The Find phase queries the objects in your S3 data lakes to find
+     any objects which contain records where a specified column contains at
+     least one of the Match IDs in the deletion queue. If any queries fail, the
+     job will abandon as soon as possible and the Forget phase will not take
+     place. The Forget Phase takes the list of objects returned from the Find
+     phase, and deletes only the relevant rows in those objects.
+   - **Optimised for Parquet:** The process of separating the two phases, as
+     opposite of processing the read/write on an object basis, is another cost
+     optimisation technique leveraging columnar dense formats such as Parquet
+     allow to efficiently find matches by reading only relevant columns instead
+     of full content from all columns. By processing only the relevant columns
+     during the Find and delegating the queries to a serverless managed service
+     such as Amazon Athena, we can then process full rows only during the Forget
+     phase, restricting the most expensive operation to only the relevant
+     objects.
```

Comment:
I find this point is a bit difficult to follow - there's a lot of info including some bits that are perhaps not necessary (e.g. describing the approach _not taken_). I don't have an immediate suggested edit, but I think simplifying the sentence to only make a few points would make it more readable.

Reply:
```
    - **Optimised for Parquet:** The split phase approach optimises scanning
      for columnar dense formats such as Parquet.  The Find phase only
      retrieves and processes the data for relevant columns when determining
      which S3 objects need to be processed in the Forget phase. This
      approach can have significant cost savings when operating on large data
      lakes with sparse matches.
```

How about this?